src/main/java/sg/ncl/MainController.java

```diff
@@ -2411,21 +2424,26 @@ public String saveExperimentImage(
             log.warn("Save image: error with exception {}", exceptionState);
 
             switch (exceptionState) {
+                case INSUFFICIENT_PERMISSION_EXCEPTION:
+                    log.warn("Imaging saving error: {}", error.getMessage());
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "Imaging saving error: {}" 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)